pyspecdata/DCCT_function.py

```diff
@@ -51,57 +46,67 @@ def DCCT(
-                    string for color mapping if specified
-    pass_frq_slice: boolean
-                    if true will show the frequency sliced out with hatching
-    just_2D:        boolean
-                    If true will only return axis coordinates/shape NOT ax_list
-    scaling_factor: float
-                    If using custom scaling this allows user to set the scaling
-                    factor
-    max_coh_jump:   dict
-                    maximum allowed transitions for each phase cycle
-    direct:         str
-                    name of direct axis
-    plot_title:     str
-                    title for DCCT plot
+    this_nddata : nddata
+        data being plotted
+    this_fig_obj : figure
+        size/type of figure to be plotted on
+    custom_scaling : boolean
+        allows user to scale the intensity of data presented
+    bbox : list
+        contains the following:
+        :bbox[0]: int
+            Left hand side padding between the left side of the figure
+            and the left side of the decorations.
+        :bbox[1]: int
+            Distance between the bottom of the figure and the bottom of the
+            lowest axes object (in Figure coordinates)
+        :bbox[2]: int
+            Distance between the right most side of the axes objects and the
+            left most side of the decorations (in Figure coordinates)
+    grid_bottom : float
+        Figure coordinates
+        y-coordinate for bottom of DCCT plot.
+    grid_top : float
+        Figure coordinates
+        y-coordinate top of grid in figure
+    top_pad : float
+        Figure coordinates
+        Distance between top of figure and top of grid
+    total_spacing : float
+        Figure coordinates
+        Spacing between phase cycle dimensions
+    vert_label_spacer : int
+        Display coordinates
+        Spacing between vertical lines that label the coherence pathways
+    shareaxis : boolean
+        Subplots scale together, but currently, this means there
+        must be tick labels on both top and bottom
+    cmap : str
+        Color mapping if specified
+    pass_frq_slice : boolean
+        If true will show the frequency sliced out with hatching
+    just_2D : boolean
+        If true will only return axis coordinates/shape NOT ax_list
+    scaling_factor : float
+        If using custom scaling this allows user to set the scaling
+        factor
+    max_coh_jump : dict
+        Maximum allowed transitions for each phase cycle
+    direct : str
+        Name of direct axis
+    plot_title : str
+        Title for DCCT plot
     """
-    x = []
-    y = []
+    x = []  # List to put direct dim into
+    y = []  # List to put indirect dims into
     my_data = this_nddata.C
     ordered_labels = {}
+
+    def ax_to_fig(thisax, thisfig):
+        """Transformation to convert axes coordinates of thisax into figure
+        coordinates of thisfig"""
+        return thisax.transAxes + thisfig.transFigure.inverted()
+
```

Comment:
There are several places where we go from axes to figure coords specifically it's used to get the coordinate of the bottom left corner of the bottom axes. It is much easier to use this as a pinpoint of reference since the coord in axes coords is 0,0